### ISSUE TITLE
GLFW sets wrong swap interval under WGL

### ISSUE THREAD

**Issue by janekb04** [glfw/glfw#1970]
```
#include <GL/glew.h>
#include <GLFW/glfw3.h>
#include <iostream>
#include <windows.h>

int main(void) {
  if (!glfwInit())
    return -1;
  glfwWindowHint(GLFW_CLIENT_API, GLFW_OPENGL_API);
  GLFWwindow *window = glfwCreateWindow(640, 480, "Hello World", NULL, NULL);
  if (!window) {
    glfwTerminate();
    return -1;
  }

  glfwMakeContextCurrent(window);

  if (glewInit() != GLEW_OK) {
    glfwTerminate();
    return -1;
  }

  std::cout << ((int (*)(void))(wglGetProcAddress("wglGetSwapIntervalEXT")))();
  glfwSwapInterval(1);
  std::cout << ((int (*)(void))(wglGetProcAddress("wglGetSwapIntervalEXT")))();

  while (!glfwWindowShouldClose(window)) {
    glClear(GL_COLOR_BUFFER_BIT);
    glfwSwapBuffers(window);
    glfwPollEvents();
  }

  glfwTerminate();
  return 0;
}
```

The code above outputs `10`. This means that the call to `glfwSwapInterval` [not only doesn't set the value to `1`](https://github.com/glfw/glfw/blob/6ed5294223fec513c3b1e95f1857b3104e2e9b86/src/wgl_context.c#L361), but sets it to `0` instead.

I think this is caused by some misunderstanding because the GLFW docs say that the default value for `glfwSwapInterval` is `0`, while [WGL docs say](https://www.khronos.org/registry/OpenGL/extensions/EXT/WGL_EXT_swap_control.txt#:~:text=The%20default%20swap%20interval%20is%201.) that the swap interval by default is `1`.

This causes an issue in my [project](https://github.com/janekb04/emscripten-test) because after calling `glfwSwapInterval(0)` once, I am unable to unset it. After that call, the window gets jittery when being resized. Calling `glfwSwapInterval(1)` does change the framerate, but doesn't solve jittering.

**Comment by NovaSmoof** [comment]
I could not replicate this. Is it still occurring @janekb04?

**Comment by janekb04** [comment]
@NovaSmoof @elmindreda This issue has been fixed by 4005f70eef41a691b849cd06e085d9005df6b66b. Indeed, now it prints `11`. After running `git revert 4005f70eef41a691b849cd06e085d9005df6b66b` it prints `10` again.

**Comment by janekb04** [comment]
At least it is fixed on Windows 10 20H2 (build `19042.1415`). I don't know what about other versions of Windows.

**Comment by elmindreda** [comment]
@janekb04 Glad to hear it! Closing this for now, then. Please post a follow-up if the issue reappears.
